### DISCUSSION TITLE
Cookies set by FastAPI not received by Next.js

### DISCUSSION THREAD

**Discussion by dev-bilal-raza** [fastapi/fastapi#11359] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [x] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
@app.post("/api/login")
def login(user_login_form: LoginForm, response: Response, session: Annotated[Session, Depends(get_session)]):
    # Call the logIn_func function to authenticate the user and obtain tokens
    user_data, access_token, refresh_token = logIn_func(
        **user_login_form.model_dump(), session=session).values()

    # Set cookies in the response with the access and refresh tokens
    response.set_cookie(
        key="access_token",
        value=access_token,
        expires=int(ACCESS_TOKEN_EXPIRE_TIME.total_seconds()),
        secure=False,  # Set to True if serving over HTTPS
        samesite="none",  # Set SameSite attribute to "None" for cross-site cookies
        httponly=True  # Prevent JavaScript access to the cookie
    )
    response.set_cookie(
        key="refresh_token",
        value=refresh_token,
        expires=int(REFRESH_TOKEN_EXPIRE_TIME.total_seconds()),
        secure=False,  # Set to True if serving over HTTPS
        samesite="none",  # Set SameSite attribute to "None" for cross-site cookies
        httponly=True  # Prevent JavaScript access to the cookie
    )

    # If user_data is not found, raise NotFoundException
    if not user_data:
        raise NotFoundException("User")

    # Return user data
    return user_data
```

### Description

I am facing an issue where cookies set by my **FastAPI** backend are not being received by my **Next.js** frontend. Although I can see that the _cookies_ are being set in the **response headers**, they are not being saved in the **browser's cookie section**. I have also set up middleware in FastAPI to handle **CORS**, but the problem persists.

here is my backend code:

![code](https://github.com/tiangolo/fastapi/assets/140187049/ade77ca6-cb3b-401e-b7e2-f118b9384625)

here is a preview of response headers:

![brwoser](https://github.com/tiangolo/fastapi/assets/140187049/2dd31d7b-fd52-467f-b0c3-11bbe2a2bec7)

here is my browser's cookie section:

![cookie in browser](https://github.com/tiangolo/fastapi/assets/140187049/7a47ad1a-c1d2-486f-8bc0-a3545ffc3e7d)

Could someone please provide guidance on how to troubleshoot and resolve this issue? **_Thank you_**.

### Operating System

Windows

### FastAPI Version

0.110.0

### Pydantic Version

2.6.4

### Python Version

3.12

**Comment by pythonweb2** [comment]
Doesn't look like a FastAPI issue here. Probably something to do with your browser settings. Here is a quick example that I used and it set the cookie as expected (using Edge)

```python
from fastapi import FastAPI, Response
from pydantic import BaseModel

app = FastAPI()


class Test(BaseModel):
    username: str
    password: str


@app.post("/test")
async def post_test(
    response: Response,
    body: Test,
) -> Test:
    response.set_cookie(
        key="access_token",
        value="asdf123",
        expires=60,
        secure=False,  # Set to True if serving over HTTPS
        samesite="none",  # Set SameSite attribute to "None" for cross-site cookies
        httponly=True,  # Prevent JavaScript access to the cookie
    )
    return body

```

**Comment by omarcruzpantoja** [comment]
Probably not a FastAPI issue. 
If you're working with two separate domains you need to tackle that. (by default some browsers will not allow cookies to be set if the domains are different). make sure that you're not getting any `console` errors or warnings when receiving the api call.

**Reply by dev-bilal-raza** [reply, marked as answer]
I think this is happening because of the Chrome browser, because when I run this code in the Edge browser, it works.
